cmake/NVBenchDependentDlls.cmake

```diff
@@ -0,0 +1,50 @@
+# By default, add dependent DLLs to build/install dirs on MSVC. This avoids
+# a variety of runtime issues when using NVML, etc.
+# This behavior can be disabled using the following options:
+if (WIN32)
+  option(NVBench_ADD_DEPENDENT_DLLS_TO_BUILD
+    "Copy dependent dlls to NVBench library build location (MSVC only)."
+    ON
+  )
+  option(NVBench_ADD_DEPENDENT_DLLS_TO_INSTALL
+    "Copy dependent dlls to NVBench library install location (MSVC only)."
+    ON
+  )
+else()
+  # These are forced off for non-MSVC builds, as $<TARGET_RUNTIME_DLLS:...>
+  # will always be empty on non-dll platforms.
+  set(NVBench_ADD_DEPENDENT_DLLS_TO_BUILD OFF)
+  set(NVBench_ADD_DEPENDENT_DLLS_TO_INSTALL OFF)
+endif()
+
+if (NVBench_ADD_DEPENDENT_DLLS_TO_BUILD OR
+    NVBench_ADD_DEPENDENT_DLLS_TO_INSTALL)
+  message(STATUS
+    "CMake 3.21.0 is required when NVBench_ADD_DEPENDENT_DLLS_TO_BUILD "
+    "or NVBench_ADD_DEPENDENT_DLLS_TO_INSTALL are enabled."
+  )
+  cmake_minimum_required(VERSION 3.21.0)
+endif()
+
+function(nvbench_setup_dep_dlls target_name)
+  # The custom command below fails when there aren't any runtime DLLs to copy,
+  # so only enable it when a relevant dependency is enabled:
+  if (NOT NVBench_ENABLE_NVML)
+    return()
+  endif()
+
+  if (NVBench_ADD_DEPENDENT_DLLS_TO_BUILD)
+    add_custom_command(TARGET ${target_name}
+      POST_BUILD
+      COMMAND
+        "${CMAKE_COMMAND}" -E copy
+          $<TARGET_RUNTIME_DLLS:${target_name}>
+          $<TARGET_FILE_DIR:${target_name}>
+      COMMAND_EXPAND_LISTS
+    )
+  endif()
+
+  if (NOT NVBench_ADD_DEPENDENT_DLLS_TO_INSTALL)
+    # TODO -- not sure how to best handle this.
```

Comment:
@robertmaynard This is another trouble spot -- would a naive install step be okay here, or would this need to be aware of rapids-cmake export sets?

Reply:
I don't think you want to install the nvml dlls with nvbench. I expect that will cause problems when shipped to a different machine running a different cuda driver version.

I think you just want to establish the correct targets so that consumers of nvbench can copy the correct dll's for build directories.